docs/en/sql-reference/functions/string-functions.md

```diff
@@ -1392,9 +1364,7 @@ Alias: `trim(input_string)`.
 
 **Returned value**
 
-A string without leading and trailing common whitespaces.
-
-Type: `String`.
+A string without leading and trailing common whitespaces. [String](../data-types/string.md).
```

Comment:
I guess we should be consistent if we prepend the return value description with `-` or not. Here and above we don't but everywhere else we do.

Perhaps we can do this:
- if a single value is returned (99%) of all cases, we omit `-`
- if a composite value (e.g. tuple, array) is returned, we add `-`

Reply:
I like the idea, although considering we don't have any automated style on formatting for functions yet and many different people contributing documentation to the project, we should maybe try keep it as simple as possible. I propose to go through and make sure all functions have a single `-`. That way it's also consistent with the arguments which have one `-` per argument.

Reply:
Sounds like a good plan.

But please let's do this in a separate PR, this one is already big enough to make my browser stutter when I review :-)

Reply:
By the way, I saw in some cpp files for functions that examples and descriptions are occasionally included in `REGISTER_FUNCTION()` eg. 
https://github.com/ClickHouse/ClickHouse/blob/f21c7505a21e6879c083086835e390019da3621a/src/Functions/mortonEncode.cpp#L339-L343
Do you know what happens with these currently? I wondered if it would be possible to extend that functionality such that when a new function gets added the contributor adds the information there (function name, description, category, arguments, return types) and we then have a github action with a script that generates the markdown and tags a maintainer from the docs team to check the spelling and fix any markdown issues. That way we can ensure a greater level of consistency in future.

Reply:
What you described is how function documentation is eventually supposed to work.

Historically, functions have been documented only via markdown. That had the problem that 1. documentation was not always in-sync with the actual implementation (usually because the programmer updating the implementation forgot updating the docs). 2. Some functions had no markdown docs at all (which you address right now).

Thus came the idea for in-source documentation (`FunctionDocumentation` in C++), which is also shown by system table `sytem.functions`. New functions are required to provide in-source docs (this is checked in CI). Programmers are also encouraged to add in-source docs to existing functions, but still the majority of existing functions is still w/o such docs.

Having markdown and in-source docs creates another problem: keeping both in-sync. The only real solution is to have one single-point of truth, i.e. keep in-source docs and generate markdown from them. Unfortunately, nobody did that yet. This requires:
- attention to detail how special markdown formatting can be represented in C++, e.g. links, `:::note`, LaTeX formula (yes, we sometimes use them in markdown docs), etc.
- an integration of the markdown generator script with CI (e.g. via nightly runs).
- an effort for _each function_ to merge both doc variants into one variant (in-source) and mark it somehow "in-source documented only". The effort will be incremental as doing it for all 100s of functions at once is infeasible.

In any case, before going that route we should surely finish the current documenting-the-undocumented project.